filebeat/prospector/prospector.go

```diff
@@ -159,8 +159,8 @@ func (p *Prospector) Run(once bool) {
 // All state updates done by the prospector itself are synchronous to make sure not states are overwritten
 func (p *Prospector) updateState(event *input.Event) error {
 
-	// Add ttl if cleanOlder is enabled
-	if p.config.CleanInactive > 0 {
+	// Add ttl if cleanOlder is enabled and TTL is not already 0
+	if p.config.CleanInactive > 0 && event.State.TTL != 0 {
```

Comment:
I'm wondering if the new check should be for `event.State.TTL > 0` since event.State.TTL starts as a `-1 * time.Second`.